tools/python/upload_and_run_browserstack_tests.py

```diff
@@ -0,0 +1,93 @@
+import argparse
+import json
+import time
+
+import requests
+
+parser = argparse.ArgumentParser("Upload and run BrowserStack tests")
+
+parser.add_argument("--id", type=str, help="BrowserStack user ID")
+parser.add_argument("--token", type=str, help="BrowserStack token")
+parser.add_argument("--test_platform", type=str, help="Testing platform, i.e., XCUITest or Espresso")
+parser.add_argument("--app_apk_path", type=str, help="Path to the app APK")
+parser.add_argument("--test_apk_path", type=str, help="Path to the test suite APK")
+parser.add_argument(
+    "--devices",
+    type=str,
+    nargs="+",
+    help="List of devices to run the tests on. For more info, "
+    "see https://www.browserstack.com/docs/app-automate/espresso/specify-devices",
+)
+
+args = parser.parse_args()
+
+
+def post_response_to_json(response):
+    if len(response) == 0:
+        raise Exception("No response from BrowserStack")
+    try:
+        json_response = json.loads(response)
+        print("JSON response: ", json.dumps(json_response, indent=2))
+        if "error" in json_response:
+            raise Exception(json_response["error"])
+        return json_response
+    except Exception as e:
+        print("response received: ", response)
+        raise Exception("Invalid JSON response from BrowserStack") from e
+
+
+def upload_apk_parse_json(post_url, apk_path):
+    with open(apk_path, "rb") as apk_file:
+        response = requests.post(post_url, files={"file": apk_file}, auth=(args.id, args.token))
```

Comment:
Should we configure specific timeouts based on the operation? I can imagine apk upload may take longer than what the default timeout (assuming there is one) allows for.

Reply:
Oh that's a good point -- I added a timeout for the entire script of 15 minutes, but I'm not sure if adding specific timeouts for the distinct steps would be better